src/atlantis/common/rpc.go

```diff
@@ -66,6 +71,20 @@ func (r *RPCClient) newClient(region int) (*rpc.Client, error) {
 	return rpc.DialHTTP("tcp", r.Opts[region].RPCHostAndPort())
 }
 
+func (r *RPCClient) newClientOnChannel(region int) chan *ClientResult {
+	c := make(chan *ClientResult)
+	go func() {
+		if r.UseTLS {
+			client, err := r.newTLSClient(region)
+			c <- &ClientResult{client: client, err: err}
+		} else {
+			client, err := rpc.DialHTTP("tcp", r.Opts[region].RPCHostAndPort())
+			c <- &ClientResult{client: client, err: err}
+		}
+	}()
+	return c
```

Comment:
Hmm?  This function always returns a channel, and the value in the channel includes the error from the call which could potentially fail.  I'm not sure what "err" you're referring to...